java/rocksjni/sst_file_metadata.cc

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+//
+// This file implements the "bridge" between Java and C++ for
+// ROCKSDB_NAMESPACE::FilterPolicy.
+
+#include <jni.h>
+
+#include "include/org_rocksdb_SstFileMetaData.h"
+#include "rocksdb/metadata.h"
+#include "rocksjni/portal.h"
+
+/*
+ * Class:     org_rocksdb_SstFileMetaData
+ * Method:    newSstFileMetaData
+ * Signature: ()J
+ */
+jlong Java_org_rocksdb_SstFileMetaData_newSstFileMetaData__(JNIEnv*, jclass) {
+  auto* metadata = new ROCKSDB_NAMESPACE::SstFileMetaData();
+  return reinterpret_cast<jlong>(metadata);
+}
+
+/*
+ * Class:     org_rocksdb_SstFileMetaData
+ * Method:    newSstFileMetaData
+ * Signature: (Ljava/lang/String;Ljava/lang/String;JJJ[BI[BIJZJJ)J
+ */
+jlong Java_org_rocksdb_SstFileMetaData_newSstFileMetaData__Ljava_lang_String_Ljava_lang_String_JJJ3BI3BIJZJJ(
+    JNIEnv*, jclass, jstring jfile_name, jstring jpath, jlong jsize,
+    jlong jsmallest_seqno, jlong jlargest_seqno, jbyteArray jsmallest_key,
+    jint jsmallest_key_len, jbyteArray jlargest_key, jint jlargest_key_len,
+    jlong jnum_reads_sampled, jboolean jbeing_compacted, jlong jnum_entries,
+    jlong jnum_deletions) {
+  auto* metadata = new ROCKSDB_NAMESPACE::LiveFileMetaData();
+
+  jboolean has_exception = JNI_FALSE;
+  std::string file_name = ROCKSDB_NAMESPACE::JniUtil::copyStdString(
+      env, jfile_name, &has_exception);
+  if (has_exception == JNI_TRUE) {
+    ROCKSDB_NAMESPACE::RocksDBExceptionJni::ThrowNew(
+        env, "Could not copy jstring to std::string");
+    return;
+  }
+
+  has_exception = JNI_FALSE;
+  std::string path =
+      ROCKSDB_NAMESPACE::JniUtil::copyStdString(env, jpath, &has_exception);
+  if (has_exception == JNI_TRUE) {
+    ROCKSDB_NAMESPACE::RocksDBExceptionJni::ThrowNew(
+        env, "Could not copy jstring to std::string");
+    return;
+  }
+
+  has_exception = JNI_FALSE;
```

Comment:
I don't think you need to keep setting `has_exception` to `JNI_FALSE`. If it was true, the function will have already returned.